app/routes/metadata.rb

```diff
@@ -1,28 +1,16 @@
+require_relative '../../x'
+
 module ExercismWeb
   module Routes
     class Metadata < Core
-      get '/exercises/:track_id/:slug' do |track_id, slug|
-        status, response = Xapi.get("v2", "exercises", track_id, slug)
-        data = JSON.parse(response)
-        if status == 404
-          flash[:notice] = data['error']
-          redirect '/'
-        end
-
-        locals = Presenters::Assignment.from_json_data(data).to_locals
-        erb :"exercises/test_suite", locals: locals
+      get '/exercises/:id/:slug/tests' do |track_id, slug|
+        _, response = X::Exercise::TestFiles.find(track_id, slug)
```

Comment:
Hm. Yeah, it looks like it is the status and the response body. Let's not leak that data out. Take a look at how the `X::Track` is implemented. `find` returns a `Track` object, not http stuff.